shared/challenge_builder.py

```diff
@@ -0,0 +1,299 @@
+    elif 'geometry' in element:
+        if element['geometry']['type'] == 'Point':
+            element['lat'] = element['geometry']['coordinates'][1]
+            element['lon'] = element['geometry']['coordinates'][0]
+        elif element['geometry']['type'] == 'LineString':
+            # Create a center point as an average of all points
+            lat = 0
+            lon = 0
+            for point in element['geometry']['coordinates']:
+                lat += point[1]
+                lon += point[0]
+            element['lat'] = lat / len(element['geometry']['coordinates'])
+            element['lon'] = lon / len(element['geometry']['coordinates'])
+        elif element['geometry']['type'] == 'Polygon':
+            # Create a center point as an average of all points
+            lat = 0
+            lon = 0
+            for point in element['geometry']['coordinates']:
+                lat += point[1]
+                lon += point[0]
+            element['lat'] = lat / len(element['geometry']['coordinates'])
+            element['lon'] = lon / len(element['geometry']['coordinates'])
+    else:
+        print(element)
+        raise ValueError("No handalable coordinates found for element")
+    return element
+
+def createGeometryFromElement(element):
+    # "geometry" needs to check before "bounds", because when you output geom, bbox will also be given back
+    # if there is "bounds" in element, construct a polygon with 4 points that represents the bbox
+    if "lat" in element:
+        # list of 2 floats for a dict that has lat lon as keys
+        element["simpleGeometry"] = [element["lon"], element["lat"]]
+    elif "geometry" in element and "coordinates" in element["geometry"]:
+        # list of lists of 2 floats for a dict that has lat lon as keys
+        element["simpleGeometry"] = element["geometry"]["coordinates"]
+    elif "geometry" in element:
+        # list of lists of 2 floats for a dict that has lat lon as keys
+        element["simpleGeometry"] = [[point["lon"], point["lat"]] for point in element["geometry"]]
+    elif "bounds" in element:
+        element["simpleGeometry"] = [
+            [element["bounds"]["minlon"], element["bounds"]["minlat"]],
+            [element["bounds"]["minlon"], element["bounds"]["maxlat"]],
+            [element["bounds"]["maxlon"], element["bounds"]["maxlat"]],
+            [element["bounds"]["maxlon"], element["bounds"]["minlat"]],
+            [element["bounds"]["minlon"], element["bounds"]["minlat"]]
+        ]	
+    else:
+        print(element)
+        raise ValueError("No handalable geometry found for element")
+    return element
+
+def getElementCenterPoint(element):
+    newElement = createElementCenterPoint(element)
+    return [newElement["lon"], newElement["lat"]]
+
+def getElementGeometry(element):
+    newElement = createGeometryFromElement(element)
+    return newElement["simpleGeometry"]
```

Comment:
Should all be obsolete.
The working principle should always be:
1. convert Overpass results to geometry / geojson instance
2. Manipulate the geometry/Create a geometry if needed via turf package functions
3. Pass the desired geometry to the GeoFeature